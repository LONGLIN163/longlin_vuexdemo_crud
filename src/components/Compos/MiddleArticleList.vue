<template>
  <el-container>
    <el-main>
      <el-table
        :data="filteredList.filter(data => !search || data.name.toLowerCase().includes(search.toLowerCase()))"
        :default-sort = "{prop: 'id', order: 'descending'}"
        style="width: 100%"
      >
        <el-table-column type="expand">
          <template slot-scope="props">
            <el-row :gutter="20">
              <el-col :span="12">
                <div class="grid-content bg-purple">
                  <el-form label-position="left" inline class="demo-table-expand-left" >
                    <el-form-item label="ID:">
                      <span>{{ props.row.id }}</span>
                    </el-form-item>
                    <el-form-item label="Name:">
                      <span>{{ props.row.name }}</span>
                    </el-form-item>
                    <el-form-item label="Taste:">
                      <span>{{ props.row.taste }}</span>
                    </el-form-item>
                    <el-form-item label="Color:">
                      <span>{{ props.row.color }}</span>
                    </el-form-item>
                    <el-form-item label="Expires date:">
                      <span>{{ props.row.expires }}</span>
                    </el-form-item>
                    <el-form-item label="Price:">
                      <span>{{ props.row.price }}</span>
                    </el-form-item>

                  <el-button 
                      size="mini" 
                      type="danger" 
                      data-test="delArticleOnTable"
                      @click="delArticleAction(props.row.id)"
                    >Delete</el-button>

                  </el-form>
                </div>
              </el-col>
              <el-col :span="12">
                <div class="grid-content bg-purple">
                  <el-form label-position="left" inline class="demo-table-expand-right" >
                    <el-form-item label="Image:">
                      <span>
                        <el-image 
                          style="width: 400px;"
                          :src="props.row.image"
                        ></el-image>
                      </span>
                    </el-form-item>
                  </el-form>
                </div>
              </el-col>
            </el-row>
          </template>
        </el-table-column>

        <el-table-column
          label="Article ID"
          sortable
          prop="id">
        </el-table-column>
        <el-table-column
          label="Name"
          title='name'
          id='name'
          sortable
          prop="name">
        </el-table-column>
        <el-table-column
          label="Taste"
          sortable
          prop="taste">
        </el-table-column>
        <el-table-column
          label="Expires date"
          sortable
          prop="expires">
        </el-table-column>
        <el-table-column
          label="Price"
          sortable
          prop="price">
        </el-table-column>

        <el-table-column align="right">
          <template slot="header" slot-scope="scope">
            <el-input
              v-model="search"
              size="mini"
              placeholder="search..."/>
          </template>

          <template slot-scope="scope">
            <el-button
              size="mini"
              type="danger"
              data-test="delArticle"
              @click="delArticleAction(scope.row.id)"
              >Delete</el-button>
          </template>
        </el-table-column>
      </el-table>

      <Dialog 
        :formMode='formMode'
        :form='form'
        v-if="dialogFormVisible"
      ></Dialog>
      
      <el-header style="text-align: center; font-size: 12px; margin-top: 16px;">
        <el-button id="createBtn" type="primary" round @click="TOOGLE_DIALOGFORM">Create Article</el-button>
      </el-header>
      
    </el-main>
  </el-container>
</template>

<script>

    import Dialog  from '@/components/Compos/Dialog.vue'
    import {mapState,mapMutations, mapActions} from 'vuex'

    export default {
        data() {
          return {
            formMode:'', 
            form: {
                name: '',
                description: '',
                taste: '',
                color: '',
                image: '',
                isFruit: true,
                expires: '',
                price: ''
            },
            search: ''
          }
        },
        components:{
          Dialog
        },
        computed:{
          ...mapState(['dialogFormVisible','filteredList'])
        },
        methods: {
          ...mapMutations(['SHOW_SELECTED_ARTICLE','TOOGLE_DIALOGFORM']),
          ...mapActions(['delArticleAction'])
        }
    }
</script>

<style lang="scss" scoped>

  .demo-table-expand-left {
    padding-left: 50px;
  }

  .demo-table-expand-right {
    padding-left: 50px;
    border-left: 1px dashed $color-font-defalt;
  }
  .demo-table-expand-left label {
    width: 90px;
    color: $color-font-defalt;
    font-weight: bold;
  }
  .demo-table-expand-left .el-form-item {
    margin-right: 0;
    margin-bottom: 0;
    width: 100%;
  }

  #createBtn{
    background-color: $color-primary;
    border: none;
  }

  .descending{
    border-top-color: $color-hint !important;
  }
    


</style>