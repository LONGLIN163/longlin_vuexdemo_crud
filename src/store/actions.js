import JSONPath from "JSONPath"
import Article from "../apis/Article";

export const setArticleListAction = ({commit})=>{
  Article.getArticleData()
  .then(res=>{
    if(res.status==200){
      const objArr=JSONPath({
        json:res.data.data,
        path:'$..[?(@.isFruit&&@.name)]'
      })
        commit('getArticleList',objArr) 
        commit('getTypes') 
    }
  }).catch(err=>{
      console.log(err)
  })
}

export const createArticleAction = ({dispatch},aobj) => {
  Article.postArticleData(aobj)
  .then(res=>{
    if(res.status==201){
      dispatch('setArticleListAction')
    }
  }).catch(err=>{
        console.log(err)
  })
}

export const delArticleAction = ({dispatch,commit},id) => {
    Article.deleteArticleData(id).then(res=>{
        if(res.status==200){
          dispatch('setArticleListAction')
          commit('ToogleRefreshCompo',true)
        }
    }).catch(err=>{
        console.log(err)
    })
}
