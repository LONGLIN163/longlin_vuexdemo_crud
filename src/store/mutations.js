
export const GET_ARTICLELIST=(state,payload)=>{
    state.articleList=payload
    state.filteredList=state.articleList
}

export const GET_TYPES=(state)=>{
    state.types=Array.from(new Set(state.articleList.map(a => a.taste)));
}

export const TOOGLE_REFRESHCOMPO=(state,refreshStatus)=>{
    state.reFresh=refreshStatus;
}

export const GET_LIST_BYTYPE=(state,typeString)=>{
    state.filteredList=state.articleList.filter((item) => {
      return item.taste===typeString
    })
}
export const RESET_FILTEREDLIST=(state)=>{
    state.filteredList=state.articleList
}
export const SHOW_SELECTED_ARTICLE=(state,val)=>{
    state.previewedProduct=val
}

export const TOOGLE_DIALOGFORM=(state)=>{
    state.dialogFormVisible=!state.dialogFormVisible
}