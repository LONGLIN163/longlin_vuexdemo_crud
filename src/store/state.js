export default {
    articleList: [],
    filteredList:[],
    dialogFormVisible:false,
    types:[],
    reFresh:false
}