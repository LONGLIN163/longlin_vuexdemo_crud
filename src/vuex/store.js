
import Vue from 'vue'
import Vuex from 'vuex'
//import Apis from '@/apis/api.js'
import axios from "axios"
import JSONPath from "JSONPath"
let apiUrl="http://localhost:3000/fruit";

Vue.use(Vuex)

const store = new Vuex.Store({
  state: {
    articleList: [],
    previewedProduct:{
      id: 0,
      name: '',
      description: '',
      taste: '',
      color: '',
      image: '',
      isFruit: true,
      expires: '',
      price: ''
    },
    dialogFormVisible:false,
    //formMode:''
  },
  //ArticleList=Al
  mutations: {
    getAlData:(state,payload)=>{
      state.articleList=payload
      console.log("init List---",state.articleList)
    },
    delArticle:(state,aId)=>{
      state.articleList=state.articleList.filter((item)=>{
        return item.articleId!==aId
      })
      console.log("After delArticle List---",state.articleList)
    },
    showSelectArticle:(state,val)=>{
      state.previewedProduct=val
      console.log("After select an item---",state.previewedProduct)
    },
    createArticle:(state,formobj)=>{
      const newId=state.articleList.reduce((p,v) => p.id < v.id ? v : p).id
      const newArticle={
        ...formobj,
        id:newId+1
      }
      state.articleList.push(newArticle)
      console.log("After createArticle List---",state.articleList)
    },
    updateArticle:(state,updatedArticle)=>{
      console.log("After updateArticle List---",state.articleList)
    },
    toggleDialogForm:(state)=>{
      state.dialogFormVisible=!state.dialogFormVisible
    }
  },
  actions:{

    setAlAction:({commit})=>{
        axios({
            url:apiUrl,
            method:"GET",
        }).then(res=>{
            if(res.status==200){
              //console.log("data***",res.data.data) 
              const objArr=JSONPath({
                json:res.data.data,
                path:'$..[?(@.isFruit&&@.name)]'
              })
               commit('getAlData',objArr) 

            }
        }).catch(err=>{
            console.log(err)
        })
    }
  }
})

export default store; 