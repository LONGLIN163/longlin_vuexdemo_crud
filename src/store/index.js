import Vue from 'vue'
import Vuex from 'vuex'
import state from './state'
import * as mutations from './mutations'
import * as actions from './actions'
import axios from "axios"
import JSONPath from "JSONPath"
let apiUrl="http://localhost:3000/fruit";

Vue.use(Vuex)

const store = new Vuex.Store({
  state,
  mutations,
  actions
})

export default store; 