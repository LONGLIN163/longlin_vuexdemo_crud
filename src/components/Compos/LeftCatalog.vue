<template>
    <el-aside width="200px" style="background-color: rgb(238, 241, 246)">
      <el-menu 
        :default-active="activeIndex" 
        class="el-menu-demo" 
        mode="vertical" 
        @select="handleSelect"
      >
          <el-menu-item index='0' :key="0">All</el-menu-item>
          <el-menu-item v-for="(item,index) in types" :index="(index+1).toString()" :key="item+index">{{item}}</el-menu-item>
      </el-menu>
    </el-aside>
</template>

<script>
    import store  from '@/vuex/store'
    import {mapState,mapMutations} from 'vuex'
    export default {
        data() {
          return {
            activeIndex:'0'
          }
        },
        computed:{
          ...mapState(['types'])
        },
        methods:{

        handleSelect(key) {
          //console.log("key---",key);
          this.getListByType(this.types[key-1])
        },
          ...mapMutations(['getListByType'])
        },
        store
    }
</script>

<style scoped>
  .el-aside {
    color: #333;
  }
</style>