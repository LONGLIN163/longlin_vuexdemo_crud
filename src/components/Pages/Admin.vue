<template>
  <div>
    <el-container style="border: 1px solid #eee">

        <LeftCatalog></LeftCatalog>
        <MiddleArticleList></MiddleArticleList>
        <RightArticleDetail></RightArticleDetail>

    </el-container>
  </div>
</template>

<script>

  import LeftCatalog from '@/components/Compos/LeftCatalog'
  import MiddleArticleList from '@/components/Compos/MiddleArticleList'
  import RightArticleDetail from '@/components/Compos/RightArticleDetail'

    export default {
      data() {
        return  {
          msg:"vuex:"
      }
      },
      components:{
        LeftCatalog,
        MiddleArticleList,
        RightArticleDetail
      }
    }

</script>

<style scoped>

</style>