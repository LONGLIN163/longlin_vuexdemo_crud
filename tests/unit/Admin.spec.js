import { mount,createLocalVue} from '@vue/test-utils'
import Admin from '@/components/Pages/Admin'
import LeftCatalog from '@/components/Compos/LeftCatalog'
import MiddleArticleList from '@/components/Compos/MiddleArticleList'
import store  from '@/store'

import ElementUI from "element-ui"
const localVue = createLocalVue();
localVue.use(ElementUI);

describe('Page admin tests:', () => {

  const wrapper = mount(Admin,{
      localVue,
      store,
  })
 

  // check snapshot
  it('Get admin snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Admin compo should have header title', () => {
    expect(wrapper.find('.el-header').text()).toBe("longlin_vuexdemo_CR(no U)D")
  })

   // check compo rendering
  it('MiddleArticleList compo should be rendered', () => {
    expect(wrapper.findComponent(MiddleArticleList).exists()).toBe(true)
  })

  it('MiddleArticleList compo should have create btn', () => {
    expect(wrapper.find('#createBtn').text()).toBe("Create Article")
  })

  // check compo rendering
  it('LeftCatalog compo should be rendered', () => {
    expect(wrapper.findComponent(LeftCatalog).exists()).toBe(true)
  })

  it('LeftCatalog compo should have create btn', () => {
    expect(wrapper.find('#tasteTypes').text()).toBe("Flavours")
  })

})