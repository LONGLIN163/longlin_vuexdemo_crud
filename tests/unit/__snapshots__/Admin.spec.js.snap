// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page admin tests: Get admin snapshot 1`] = `
<div>
  <section class="el-container is-vertical" style="border: 1px solid #eee;">
    <header class="el-header" style="height: 60px;">longlin_vuexdemo_CR(no U)D</header>
    <section class="el-container">
      <aside class="el-aside" style="width: 200px;">
        <div id="tasteTypes">Flavours</div>
        <transition-stub mode="out-in" class="el-menu-demo">
          <ul role="menubar" class="el-menu">
            <li role="menuitem" tabindex="-1" class="el-menu-item is-active" style="padding-left: 20px;">All</li>
          </ul>
        </transition-stub>
      </aside>
      <section class="el-container">
        <main class="el-main">
          <div class="el-table el-table--fit el-table--scrollable-x el-table--enable-row-hover" style="width: 100%;">
            <div class="hidden-columns">
              <div></div>
              <div></div>
              <div title="name" id="name"></div>
              <div></div>
              <div></div>
              <div></div>
              <div></div>
            </div>
            <div class="el-table__header-wrapper">
              <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 528px;">
                <colgroup>
                  <col name="el-table_1_column_1" width="48">
                  <col name="el-table_1_column_2" width="80">
                  <col name="el-table_1_column_3" width="80">
                  <col name="el-table_1_column_4" width="80">
                  <col name="el-table_1_column_5" width="80">
                  <col name="el-table_1_column_6" width="80">
                  <col name="el-table_1_column_7" width="80">
                </colgroup>
                <thead class="">
                  <tr class="">
                    <th colspan="1" rowspan="1" class="el-table_1_column_1   undefined el-table__expand-column  is-leaf el-table__cell">
                      <div class="cell"></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_2 descending    is-leaf is-sortable el-table__cell">
                      <div class="cell">Article ID<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf is-sortable el-table__cell">
                      <div class="cell">Name<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_4     is-leaf is-sortable el-table__cell">
                      <div class="cell">Taste<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_5     is-leaf is-sortable el-table__cell">
                      <div class="cell">Expires date<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_6     is-leaf is-sortable el-table__cell">
                      <div class="cell">Price<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                    </th>
                    <th colspan="1" rowspan="1" class="el-table_1_column_7  is-right   is-leaf el-table__cell">
                      <div class="cell">
                        <div class="el-input el-input--mini">
                          <!----><input type="text" autocomplete="off" placeholder="search..." class="el-input__inner">
                          <!---->
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </div>
                    </th>
                  </tr>
                </thead>
              </table>
            </div>
            <div class="el-table__body-wrapper is-scrolling-left">
              <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 528px;">
                <colgroup>
                  <col name="el-table_1_column_1" width="48">
                  <col name="el-table_1_column_2" width="80">
                  <col name="el-table_1_column_3" width="80">
                  <col name="el-table_1_column_4" width="80">
                  <col name="el-table_1_column_5" width="80">
                  <col name="el-table_1_column_6" width="80">
                  <col name="el-table_1_column_7" width="80">
                </colgroup>
                <tbody>
                  <!---->
                </tbody>
              </table>
              <div class="el-table__empty-block" style="height: 100%; width: 528px;"><span class="el-table__empty-text">暂无数据</span></div>
              <!---->
            </div>
            <!---->
            <!---->
            <!---->
            <!---->
            <div class="el-table__column-resize-proxy" style="display: none;"></div>
          </div>
          <!---->
          <header class="el-header" style="height: 60px; text-align: center; font-size: 12px; margin-top: 16px;"><button type="button" class="el-button el-button--primary is-round" id="createBtn">
              <!---->
              <!----><span>Create Article</span>
            </button></header>
        </main>
      </section>
    </section>
  </section>
</div>
`;
